### DISCUSSION TITLE
Case insensitive query with Lucid (PostgresSQL)

### DISCUSSION THREAD

**Discussion by kaposke** [adonisjs/core#1449] [Help] [answered]
Hey!

I'm using PostgresSQL, and I'm trying to do a case insensitive **where**. Here is what would work for MySQL:
`query.where('name', 'like', '%${term}%')`

A common solution on a raw query would be to do:
`SELECT * FROM table WHERE LOWER(name) LIKE LOWER('%Kaposke%')`
but I'm trying to avoid raw queries as they are unsafe.

I got around the problem using Regular Expressions, but I really don't like the looks of it, and there is probably a better way to do it.
`query.where('name', '~*', '.*${term}.*')`

Thank you in advance.

**Comment by thetutlage** [comment]
Nope, raw queries are equally safe. You just need to use bind parameters when defining a raw where clause. Infact, the core of the framework uses the `lower` method. Here's the source https://github.com/adonisjs/lucid/blob/develop/src/Bindings/Validator.ts#L168

**Comment by lucasdoamaral** [comment, marked as answer]
Since you're using PostgreSQL you could use the insensitive operator **ilike**:

`query.where('name', 'ilike', '%${term}%')`

**Reply by kaposke** [reply]
Marking this as the answer instead, since it was the part I was missing about Postgres. Thanks your reply.

**Reply by roneydourados** [reply]
Hey, thanks, your saved my life!! kkkkk

**Reply by gusflopes** [reply]
just got saved by this also. lol

**Reply by georgebsmith-tech** [reply]
And saved me also
